### PR TITLE
[new module] Azure CDN endpoint module

### PR DESCRIPTION
**Pull request by caohai**
- add azure_rm_cdnendpoint module
- add integration test for azure_rm_cdnendpoint

### CODE REVIEW

File: library/azure_rm_cdnendpoint.py [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_cdnendpoint
+version_added: "2.7"
+short_description: Manage a CDN endpoint.
```
Comment (yungezz) [inline library/azure_rm_cdnendpoint.py]: Azure CDN

File: library/azure_rm_cdnendpoint.py [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+description:
+    - Create, update, start, stop and delete a CDN endpoint.
```
Comment (yungezz) [inline library/azure_rm_cdnendpoint.py]: Azure

File: library/azure_rm_cdnendpoint.py [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+        default: present
+        choices:
+            - absent
+            - present
+    location:
+        description:
+            - Valid azure location. Defaults to location of the resource group.
+    started:
+        description:
+            - Use with state 'present' to start the machine. Set to false to have the endpoint be 'stopped'. Set to true to start the endpoint.
```
Comment (yungezz) [inline library/azure_rm_cdnendpoint.py]: machine?

File: library/azure_rm_cdnendpoint.py [hunk 4]
```diff
@@ -0,0 +1,623 @@
...
+          name: TestEndpoint
+          profile_name: TestProfile
+          state: absent
+'''
+RETURN = '''
+state:
+    description: Current state of the CDN endpoint
+    returned: always
+    type: dict
+    example:
```
Comment (yungezz) [inline library/azure_rm_cdnendpoint.py]: example

File: library/azure_rm_cdnendpoint.py [hunk 5]
```diff
@@ -0,0 +1,623 @@
...
+
+        resource_group = self.get_resource_group(self.resource_group)
+        if not self.location:
+            self.location = resource_group.location
+
+        response = self.get_cdnendpoint()
+
+        if self.state == 'present':
+
+            if not response and self.started is None:
```
Comment (yungezz) [inline library/azure_rm_cdnendpoint.py]: the if/elif seems little confusing. why not
if not response:
     if self.started is None:
     else:
elsle:

File: library/azure_rm_cdnendpoint.py [hunk 6]
```diff
@@ -0,0 +1,623 @@
...
+        :return: deserialized Azure CDN endpoint instance state dictionary
+        '''
+        self.log("Creating the Azure CDN endpoint instance {0}".format(self.name))
+
+        parameters = Endpoint(
+            origins=[
+                DeepCreatedOrigin(
+                    name=self.origin['name'],
+                    host_name=self.origin['host_name'],
+                    http_port=self.origin['http_port'],
```
Comment (yungezz) [inline library/azure_rm_cdnendpoint.py]: none check? http_port and https_port is not  required

File: library/azure_rm_cdnendpoint.py [hunk 7]
```diff
@@ -0,0 +1,623 @@
...
+    def delete_cdnendpoint(self):
+        '''
+        Deletes the specified Azure CDN endpoint in the specified subscription and resource group.
+
+        :return: True
+        '''
+        self.log("Deleting the CDN endpoint {0}".format(self.name))
+        try:
+            poller = self.cdn_management_client.endpoints.delete(
+                self.resource_group, self.profile_name, self.name)
```
Comment (yungezz) [inline library/azure_rm_cdnendpoint.py]: a side question, if delete cdn profile, will cdn endpoints be deleted?

File: tests/integration/targets/azure_rm_cdnendpoint/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+      origin:
+        name: "org{{ endpointname }}"
+        host_name: "www.google.com"
+      tags:
+          testing: testing
+          delete: on-exit
+          foo: bar
+  register: output
+
+- name: Assert the Traffic Manager profile is well created
```
Comment (yungezz) [inline tests/integration/targets/azure_rm_cdnendpoint/tasks/main.yml]: typo, traffic manager

File: library/azure_rm_cdnendpoint.py (hunk 7 above)
Comment (caohai) [reply library/azure_rm_cdnendpoint.py]: Yes. Endpoints are affiliated to a specific CDN profile. Profile is similar to a resource group, once deleted, all endpoints in that profile are gone. (Tested in portal and Ansible)

File: library/azure_rm_cdnendpoint.py (hunk 6 above)
Comment (caohai) [reply library/azure_rm_cdnendpoint.py]: In DeepCreatedOrigin constructor, http_port and https_port are set to None if not provided. In this case, service will set a default value itself(80 for http_port and 443 for https_port).
Should we set the default value in the module or let the service do the work?

**Pull request merged**